app/models/ZkKafka.scala

```diff
@@ -148,4 +160,14 @@ object ZkKafka {
 
     (new Totals(total, kafkaTotal, spoutTotal), deltas)
   }
+
+  // Attempt to parse a byte array as json, and catch and log any error rather than
+  // bubbling it up.
+  def tryParse(json: Array[Byte]): Option[JsValue] = try {
```

Comment:
It could, but since it's a pure function there's no compelling reason to hide it (I think you'll tend to see much less use of access modifiers in Scala, especially with pure code).  Idiomatic functional code tends to evolve towards mostly reusable pure functions and comparatively less code that is either mutable or specific to a particular class/operation.  In Java the common code would typically be put in a util class of some sort; in Scala you have too much general code to stick it all in one giant utility class, so people tend to use a few different approaches to code organization.

One possibility is to modularize by package name by concern, and to put all related common code in a package object.  This has the benefit that the general code is automatically visible to other classes/objects in the package.  Another is to group related functionality by file, where you end up with some mix of classes and objects with specific and general functionality grouped together.  And another is to have general code spread about a number of different utility objects or files, or some mix of differnet approaches.

I tend to use a mix of package objects and file-based grouping of related general/specfic code, with a subjective decision point when I consider the code general enough to promote out to a common utility module.  In this case, since the utility function is not yet used outside of the object in that file I left it there, and if a future refactoring required it's use in other objects I'd move it up to a package object.

Another factor is that Scala hides protected/private functions in documentation by default, which makes discovering reusable code more difficult.

I hope that explains my general thought process towards organization and why I left the function there (but visible).